cypress/integrations/admin_activities/diseases_activities/edit_disease_activity.cy.ts

```diff
@@ -0,0 +1,53 @@
+/// <reference types="cypress" />
+
+const DISEASE_START_PATH = "/admin/diseases";
+
+describe("Diseases Edit Activity specs", () => {
+  it("should render the ui", () => {
+    cy.authenticate(DISEASE_START_PATH);
+    cy.dataCy("activity-title").contains("Diseases");
+  });
+
+  it("should show disease edit form", () => {
+    cy.get('button[title="Edit"]').first().click();
+    cy.dataCy("activity-title").contains("Edit Disease");
+  });
+
+  it("should fail to edit the disease", () => {
+    cy.byId("code").should("be.disabled");
+    cy.get('input[name="opdInclude"]').uncheck();
+    cy.get('input[name="ipdInInclude"]').check();
+    cy.byId("diseaseType").clear();
+    cy.dataCy("submit-form").click();
+    cy.dataCy("dialog-info").should("not.exist");
+  });
+
+  it("should successfully save disease changes", () => {
+    cy.byId("diseaseType").click();
+    cy.get('.MuiAutocomplete-popper li[data-option-index="0"]').click();
+    cy.dataCy("submit-form").click();
+    cy.dataCy("dialog-info").contains("has been updated successfully!");
+    cy.dataCy("approve-dialog").click();
+  });
+
+  it("should redirect after disease update", () => {
+    cy.dataCy("activity-title").contains("Diseases");
+  });
+
+  it("should cancel the cancellation of the disease creation", () => {
+    cy.get('button[title="Edit"]').first().click();
```

Comment:
Since https://github.com/informatici/openhospital-ui/pull/572 Buttons can accept dataCy, maybe it's appropriate in this case to enhance readability